### PR TITLE
`Node::is_empty`

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Add a `Node::is_empty` method to replace the `uinode.size().x() <= 0. || uinode.size.y() <= 0.` checks.

### CODE REVIEW

File: crates/bevy_ui/src/render/mod.rs [hunk 1]
```diff
@@ -698,7 +698,7 @@ pub fn extract_uinode_text(
         };
 
         // Skip if not visible or if size is set to zero (e.g. when a parent is set to `Display::None`)
-        if !view_visibility.get() || uinode.size().x == 0. || uinode.size().y == 0. {
+        if !view_visibility.get() || uinode.is_empty() {
```
Comment (IQuick143) [inline crates/bevy_ui/src/render/mod.rs:701]: I assume the change in behaviour to nodes with negative extents is intentional?

**Review by IQuick143 — APPROVED** [review]

**Comment by IQuick143** [comment]
I assume NaN values need not be handled because it shouldn't even enter the code. (Else it would be good to revise the logic due to how NaN fails all comparisons.)

**Comment by ickshonpe** [comment]
> I assume NaN values need not be handled because it shouldn't even enter the code. (Else it would be good to revise the logic due to how NaN fails all comparisons.)

That's right, we shouldn't ever encounter any NaNs here.

File: crates/bevy_ui/src/render/mod.rs (hunk 1 above)
Comment (ickshonpe) [reply crates/bevy_ui/src/render/mod.rs:701]: Taffy clamps the layout size output to zero or greater, so there shouldn't be any difference between `==` or `<=`. Using `<=` in the `is_empty` seems a safer choice though.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**